routes/product.js

```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+const router = express.Router();
+
+const ProductService = require('../services/ProductService');
+const ProductServiceInstance = new ProductService();
+
+module.exports = (app) => {
+    app.use('/product', router);
+
+    router.get('/', async(req, res, next) => {
```

Comment:
ovdje imas get i getAll rute (routes ili endpoints na eng.)... trebat ce ti i create (koji koristi POST HTTP request), update (koji koristi HTTP PUT request), i delete (koji koristi DELETE HTTP request)...

predlozija bi ti da si nadjes par videa na youtube-u i procitas osnovno o REST-u... vjerojatno ste ga spominjali, al da dobro to utvrdis jer je to jedna od osnova koju ces morat razumit..

u kratkim crtama, REST (representational state transfer) je arhitekturni stil (nacin oblikovanja) web aplikacija, ili tocnije ovih endpointova (tj. ruta.. jedna te ista stvar).. 

u principu imas klijenta i server (to su tvoj frontend i backend) i oni međusobno pricaju.. 

FE aplikaciju neki korisnik "skine" (u pozadini se doslovno skine javascript datoteka) na svoj kompjuter kada otvori neku stranicu i ta aplikacija se vrti u browseru (bilo chrome, mozzila ili internet explorer, svejedno je - oni znaju kako izvodit javascript aplikacije i oni je pokrenu)

ta FE aplikacija od nekud dohvatit podatke koje ce prikazati korisniku.. i to dohvaca sa servera.. 

server je pak druga aplikacija koja je pokrenuta na nekom drugom racunalu na drugom kraju svijeta.. to moze biti doslovno neko racunalo, a moze biti i cloud (kao sto su AWS, heroku ili azure - neznam jeste to radili) i moze biti napisana u bilo kojem programu - bilo javascript, python, java itd... (s druge stranje 99% FE aplikacija su napisane u javascriptu)

FE i BE aplikacije pricaju preko interneta, koristeci HTTP protokol... taj protokol je nista drugo nego "konvencija" ili bolje receno "format" kako oblikovat poruke s kojima pricaju.. HTTP ima više "metoda" - GET, POST, PUT i DELETE su osnovne i one koje se koriste 99% vrimena

e sad.... REST je rekli smo arhitekturni stil, a napravljen je da dobro odgovara HTTP protokolu.. on nam govori kako oblikovat nase rute na BE da bi FE mogao imati interakciju sa resursima (resursi su nasi entiteti - product, order, cart, order item, user).. 

REST nam govori da uvijek za create moramo koristiti HTTP POST metodu, za get uvijek HTTP GET metodu, za update HTTP PUT metodu i za brisanje uvije HTTP DELETE metodu..

nadalje govori kako mora izgledat format ruta (endpointova)... 

GET ALL: /ime-resursa
GET: /ime-resursa/{id-resursa}
CREATE: /ime-resursa
UPDATE: /ime-resursa/{id-resursa}
DELETE: /ime-resursa/{id-resursa}

za nas resurs product na primjer bi to izgledalo ovako

GET ALL: /product
GET: /product/{id-producta} ----> na primjer /product/4 ----> dohvati product sa id-em 4, tako i za ostale
CREATE: /product
UPDATE: /product/{id-producta}
DELETE: /product/{id-producta}

jos jedna osnovna razlika je sta se kod POST i PUT HTTP metoda podatci salju u tijelu HTTP poruke (svaka HTTP poruka ima zaglavlje i tijelo, kasnije cemo i to proucit sve)

ovo je u kratko ovako iz glave, al opet cu ponovit, prouci sve sta mozes na temu REST-a (a nebi bilo lose i HTTP protokola) - to su osnove koje ces trebat razumit - dobro to naucit